src/lib/addons/datadog.ts

```diff
@@ -35,12 +47,22 @@ export default class DatadogAddon extends Addon {
             title: 'Unleash notification update',
             tags,
         };
-
+        let extraHeaders = {};
+        if (typeof customHeaders === 'string' && customHeaders.length > 1) {
+            try {
+                extraHeaders = JSON.parse(customHeaders);
+            } catch (e) {
+                this.logger.warn(
+                    'Could not parse the json in the customHeaders parameters',
```

Comment:
```suggestion
                    `Could not parse the json in the customHeaders parameters ${customHeaders}`,
```

but I think this exception should bubble up as it'd be a configuration issue

Reply:
But you wouldn't want to throw an exception when creating a feature because one of the addons failed to parse data.

Reply:
Though, this is on a different thread.